FlattenException.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\debug;
+
+use yii\base\Object;
+use yii\web\HttpException;
+
+/**
+ * FlattenException wraps a PHP Exception to be able to serialize it.
+ * Basically, this class removes all objects from the trace.
+ * Ported from Symfony components
+ *
+ * @author Dmitry Bashkarev <dmitry@bashkarev.com>
+ * @since 2.0.10
+ */
+class FlattenException extends Object
+{
+    private $_message;
+    private $_code;
+    private $_previous;
+    private $_trace;
+    private $_class;
+    private $_statusCode;
+    private $_file;
+    private $_line;
+
+    public static function create(\Exception $exception, $statusCode = null)
```

Comment:
Because you can use normal constructors? If you deviate from what is normal then you need to have an argument for doing so..